### PR TITLE
Check account age for unverify

### PR DESCRIPTION
**Pull request by tanndlin**

### CODE REVIEW

**Comment by tanndlin** [comment]
Should I take the duplicate code in `new_member_role.handler` and in `welcome.handler`?
https://github.com/joey-colon/lion/blob/ac2ba299319ca3dc7e07cc5c836faa07d0e89f09/src/app/handlers/new_member_role.handler.ts#L29-L31
https://github.com/joey-colon/lion/blob/ac2ba299319ca3dc7e07cc5c836faa07d0e89f09/src/app/handlers/welcome.handler.ts#L12-L14
And make it a function in `new_member_role.handler`, in case we wanted to tweak the account age, or just change the criteria.
and just have `const shouldUnverify = checkShouldUnverfiy(member);`

**Comment by joey-colon** [comment]
> Should I take the duplicate code in `new_member_role.handler` and in `welcome.handler`...
> https://github.com/joey-colon/lion/blob/ac2ba299319ca3dc7e07cc5c836faa07d0e89f09/src/app/handlers/new_member_role.handler.ts#L29-L31
> 
> 
> https://github.com/joey-colon/lion/blob/ac2ba299319ca3dc7e07cc5c836faa07d0e89f09/src/app/handlers/welcome.handler.ts#L12-L14
> 
> 
> And make it a function in `new_member_role.handler`, in case we wanted to tweak the account age, or just change the criteria?
> and just have `const shouldUnverify = checkShouldUnverfiy(member);`

it would make sense to create a helper method named `isVerified(member)` that both of these files reference. 

it will make sense to make a black box that encapsulates the internal logic on how we determine verification and also create only one dependency that needs to be change should we ever need to change the logic of how the method is implemented (we probably will at some point)

**Review by joey-colon — APPROVED** [review]

**Comment by joey-colon** [comment]
also, up for debate, but a simple convention would be to create a `src/app/util/member.ts` for all member related utility functions as we dont really have any established pattern atm

File: src/app/handlers/new_member_role.handler.ts [hunk 1]
```diff
@@ -26,8 +27,8 @@ export class NewMemberRoleHandler implements IHandler {
     }
     member.addRole(<Role>this._roleCache[this._UNACKNOWLEDGED_ROLE]);
 
-    const hasAvatar = Boolean(member.user.avatar);
-    if (hasAvatar) {
+    const accountIsNew = MemberUtils.shouldUnverify(member);
```
Comment (c650) [inline src/app/handlers/new_member_role.handler.ts:30]: naming logic between `accountIsNew` and `shouldUnverify` is weird, make them consistent

File: src/app/handlers/welcome.handler.ts [hunk 2]
```diff
@@ -9,8 +10,10 @@ export class WelcomeHandler implements IHandler {
   constructor(public container: IContainer) {}
 
   public async execute(member: GuildMember): Promise<void> {
-    const hasAvatar = Boolean(member.user.avatar);
-    const embed = this._createEmbed(hasAvatar);
+    // const accountIsNew = shouldUnverify(member);
+    const accountIsNew = MemberUtils.shouldUnverify(member);
```
Comment (c650) [inline src/app/handlers/welcome.handler.ts:14]: saa

File: src/app/util/member.util.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import { GuildMember, Role } from 'discord.js';
+
+export class MemberUtils {
+  //Determines whether a member should be unverified
+  //Does not specifically have to be age of account
+  public static shouldUnverify(member: GuildMember): boolean {
+    const creationDate = member.user.createdTimestamp;
+    const accountAge = creationDate / 1000 / 60 / 60 / 24; //Convert ms to days
+    return accountAge <= 30;
```
Comment (c650) [inline src/app/util/member.util.ts:9]: 2-4 days imo

**Review by c650 — APPROVED** [review]

**Pull request merged**